zio-http/jvm/src/main/scala/zio/http/netty/server/NettyDriver.scala

```diff
@@ -42,9 +45,21 @@ private[zio] final case class NettyDriver(
 
   def start(implicit trace: Trace): RIO[Scope, StartResult] =
     for {
+      boss    <- {
+        // The boss group is responsible for accepting new connections.
+        // It's recommended to use a separate, single-threaded group for this purpose to avoid contention between
+        // accepting new connections and processing I/O.
+        val cfg = nettyConfig.maxThreads(1).copy(shutdownQuietPeriodDuration = 0.seconds)
+        eventLoopGroup match {
+          case _: EpollEventLoopGroup   => EventLoopGroups.epoll(cfg)
+          case _: KQueueEventLoopGroup  => EventLoopGroups.kqueue(cfg)
+          case _: IOUringEventLoopGroup => EventLoopGroups.uring(cfg)
+          case _                        => EventLoopGroups.nio(cfg)
+        }
```

Comment:
@987Nabil another way here would be to create an `EventLoopGroups` case class that contains the boss and worker group and add it as a requirement to the ZIO environment, that way users could technically configure it. What do you think, do we have time before the 3.0.0 release to do that (since it'll be a breaking change)?

Reply:
After discussing it on Discord, we decided to go ahead and push this into a layer